### PR TITLE
Add basic touch input support for osu!mania

### PR DESCRIPTION
**Pull request by peppy**
Based on / closes #16263 (had to reopen due to the previous PR being on the user's `master` branch and not allowing commits). I've kept the original history.

Initially I went down the path of getting the `KeyBindingContainer` via (already-cached) `DrawableRuleset`, but multiple mania tests feature input managers without the `DrawableRuleset` so this felt like an okay middle-ground.

### CODE REVIEW

**Review by frenzibyte — COMMENTED** [review]
Haven't got around testing touch input yet, but mouse indicates that this is working pretty well.

File: osu.Game.Rulesets.Mania/UI/Column.cs [hunk 1]
```diff
@@ -139,5 +140,48 @@ public void OnReleased(KeyBindingReleaseEvent<ManiaAction> e)
...
+            {
+                keyBindingContainer?.TriggerPressed(column.Action.Value);
+                return base.OnMouseDown(e);
+            }
+
+            protected override void OnMouseUp(MouseUpEvent e)
+            {
+                keyBindingContainer?.TriggerReleased(column.Action.Value);
+                base.OnMouseUp(e);
+            }
```
[the remark this answers is not in the document]
Comment (peppy) [reply osu.Game.Rulesets.Mania/UI/Column.cs:173]: This is intentional and correct.

**Comment by bdach** [comment]
Tested this on an android phone. It *works*, but the columns are too narrow to comfortably play. It can't be alleviated with UI scale settings either because the playfield is unaffected by that.

But I suppose this does work for what it is and is probably the best that can be done for now. On a tablet the columns may be wide enough to comfortably click.

**Review by bdach — APPROVED** [review]

**Comment by mk56-spn** [comment]
> Tested this on an android phone. It _works_, but the columns are too narrow to comfortably play. It can't be alleviated with UI scale settings either because the playfield is unaffected by that.
> 
> But I suppose this does work for what it is and is probably the best that can be done for now. On a tablet the columns may be wide enough to comfortably click.

im guessing you played on the default skin? since thats like one of the thinnest skins ever, on something like yugens or plox its probs alot more bareable

**Review by smoogipoo — APPROVED** [review]
Lgtm

**Pull request merged**